src/Colors/index.js

```diff
@@ -0,0 +1,101 @@
+import { mergeColors, calculateAlpha, isObject, isString } from './utils.js'
+
+let colors = {
+  white: '#ffffff',
+  black: '#000000',
+  red: '#ff0000',
+  green: '#00ff00',
+  blue: '#0000ff',
+  yellow: '#feff00',
+  cyan: '#00feff',
+  magenta: '#ff00ff',
+}
+
+const normalizedColors = {
+  //store for normalized colors
+}
+
+const calculateColor = (value, options) => {
+  let targetColor = normalizeColor(value)
+  if (!isNaN(options)) {
+    options = { alpha: options }
+  }
+  if (options && isObject(options)) {
+    if (!isNaN(options.darken)) {
+      targetColor = mergeColors(targetColor, 0xff000000, 1 - options.darken)
+    }
+
+    if (!isNaN(options.lighten)) {
+      targetColor = mergeColors(targetColor, 0xffffffff, 1 - options.lighten)
+    }
+
+    if (!isNaN(options.opacity)) {
+      options.alpha = options.opacity / 100
+    }
+
+    if (!isNaN(options.alpha)) {
+      targetColor = calculateAlpha(targetColor, options.alpha)
+    }
+  }
+  return targetColor || 0
+}
+
+const normalizeColor = color => {
+  let targetColor = colors[color]
+  if (!targetColor) {
+    targetColor = color
+  }
+  if (isString(targetColor) && /^#[0-9A-F]{6}$/i.test(targetColor.toUpperCase())) {
+    targetColor = `0xff${targetColor.slice(1)}` * 1
+  }
+  return targetColor || 0xffffffff
+}
+
+const cleanUpNormalizedColor = color => {
+  for (let c in normalizedColors) {
+    if (c.indexOf(color) > -1) {
+      normalizedColors[c] = undefined
+      delete normalizedColors[c]
+    }
+  }
+}
+
+export default {
+  get(value, options) {
+    // create color tag for storage
+    let tag = `${value}${options !== undefined ? JSON.stringify(options) : ''}`
+    // check if value is not a number, or if options is not undefined
+    if (isNaN(value) || options) {
```

Comment:
do we need this check before we can return the stored color?

Reply:
No we do not